Project-II-Calculator/calculator/src/App.js

```diff
@@ -1,21 +1,91 @@
 import React from 'react';
 import './App.css';
+import NumberButton from "./components/ButtonComponents/NumberButton";
+import ActionButton from "./components/ButtonComponents/ActionButton";
+import CalculatorDisplay from "./components/DisplayComponents/CalculatorDisplay";
+
+
+// calculator's logic
+function buttonLister(e) {
+  console.log(e.toElement.innerText);
+  console.log(e.toElement.classList[1]);
+}
+
+function onClickHandler(e) {
+  const value = e.toElement.innerText;
+  const className = e.toElement.classList[1];
+  const display = document.querySelector(".display");
+
+  if (className === "action-button" || className === "number-button") {
+    const expression = display.innerText;
+    switch (value) {
+      case "=": {
+        try {
+          const result = eval(expression);
+          display.innerText = result;
+        } catch(ex) {
+          console.log(ex);
+        }
+        break;
+      }
+      case "backspace": {
+        display.innerText = expression.slice(0, expression.length - 1);
+        break;
+      }
+      case "clear": {
+        display.innerText = "";
+        break;
+      }
+      default: display.innerText += value;
+    }
+  }
+}
+
+// app start
+document.addEventListener('click', (e) => onClickHandler(e) );
```

Comment:
so in react we can add an onClick property to our html tags and just pass the function into them inside our child components that your returning in your .map function.

i encourage you to take a look at that:
https://reactjs.org/docs/handling-events.html